plenum/server/consensus/view_change_service.py

```diff
@@ -0,0 +1,57 @@
+from plenum.common.event_bus import ExternalBus
+from plenum.common.messages.node_messages import ViewChange, ViewChangeAck, NewView
+from plenum.server.consensus.consensus_data_provider import ConsensusDataProvider
+
+
+class ViewChangeService:
+    def __init__(self, data: ConsensusDataProvider, network: ExternalBus):
+        self._data = data
+        self._network = network
+
+        network.subscribe(ViewChange, self.process_view_change_message)
+        network.subscribe(ViewChangeAck, self.process_view_change_ack_message)
+        network.subscribe(NewView, self.process_new_view_message)
+
+    def start_view_change(self):
+        # TODO: Calculate
+        prepared = []
+        preprepared = []
+
+        self._data.view_no += 1
```

Comment:
I think we will need to store the view we are changing from to use it for 3PC messages from the previous view we are going to re-order.

Reply:
I think having NewView with PrePrepares from previous view would be enough. Furthermore, having a previous view_no won't do us any good since we might need to reorder batches from different previous views.